src/server/tiering/disk_storage.h

```diff
@@ -36,6 +36,7 @@ class DiskStorage {
   // Request bytes to be stored, cb will be called with assigned segment on completion. Can block to
   // grow backing file. Returns error code if operation failed  immediately (most likely it failed
   // to grow the backing file) or passes an empty segment if the final write operation failed.
+  // Bytes must not outlive the object call and can be dropped before cb is resolved
```

Comment:
`Bytes must not outlive` sounds like they have to be deleted before. Maybe just keep the second part of the sentence.